### ISSUE TITLE
Throbber is not considered as Text<'_>

### ISSUE THREAD

**Issue by Julien-cpsn** [arkbig/throbber-widgets-tui#6]
Hello!

I would really like to use this crate/lib for a personnal projet ([ATAC](https://github.com/Julien-cpsn/ATAC)) in order to make a tree component with a throbber for requests that are pending.
But the problem is that the throbber cannot be made into a `Span<'_>` nor in something like `Text<'_>`.

A great example of that is when using the [tui-tree-widget](https://docs.rs/tui-tree-widget/0.17.0/tui_tree_widget/) lib :

```rust
let throbber = Throbber::default()
    .label("some name")
    .throbber_set(throbber_widgets_tui::CLOCK)
    .use_type(throbber_widgets_tui::WhichUse::Spin);

TreeItem::new_leaf(identifier, throbber)
```

says:

```
the trait bound `Text<'_>: From<Throbber<'_>>` is not satisfied
the trait `From<Throbber<'_>>` is not implemented for `Text<'_>`

Help: the following other types implement trait `From<T>`:
<Text<'a> as From<&'a Masked<'_>>>
<Text<'a> as From<Masked<'a>>>
<Text<'a> as From<std::string::String>>
<Text<'a> as From<&'a str>>
<Text<'a> as From<Cow<'a, str>>>
<Text<'a> as From<Span<'a>>>
<Text<'a> as From<ratatui::text::Line<'a>>>
<Text<'a> as From<Vec<ratatui::text::Line<'a>>>>

Note: required for `Throbber<'_>` to implement `Into<Text<'_>>` Note: required by a bound in `TreeItem::<'a, Identifier>::new_leaf`
```

Having the throbber implementing `Span<'_>` (to go into a Line<'_>) or just `Text<'_>` would be fantastic as it would bring a bigger compatibility with other libs without denaturing the concept of this one.

Have a great day

**Comment by Julien-cpsn** [comment]
![image](https://github.com/arkbig/throbber-widgets-tui/assets/64107022/20ba50f9-d8bd-4bc6-b955-fd3d42ee9101)
Otherwise, works like a charm!

**Comment by arkbig** [comment]
@Julien-cpsn Thanks for using our library.
I have created a PR(#7) that allows `Throbber` to be converted to `Span` and `Line`.
I plan to use stateless Into traits (random indexes) and have the Throbber methods `to_symbol_span` and `to_line` as stateful.
How about something like this?

**Comment by Julien-cpsn** [comment]
@arkbig Yes that would be great!

I hope my request does not go too far from your orginal vision of the lib :)

**Comment by arkbig** [comment]
We have released v0.4.1, which adds the ability to convert to span and line.
Thank you for reporting the problems you encountered during actual use!
Please be assured that the original vision has not changed. Your feedback is invaluable to us!

**Comment by Julien-cpsn** [comment]
@arkbig I will try it soon!
TIP: put the link of the issue in the commit/release in order for issuers to see where you put the solution.
